File: Examples/test-suite/php/exception_memory_leak_runme.php
<?php

require "tests.php";

check::functions(array('trigger_internal_swig_exception'));
check::classes(array('Foo', 'exception_memory_leak'));
// No new vars
check::globals(array());

$a = new Foo();
check::equal(Foo::get_count(), 1, "Should have 1 Foo objects");
$b = new Foo();
check::equal(Foo::get_count(), 2, "Should have 2 Foo objects");

// Normal behaviour
trigger_internal_swig_exception("no problem", $a);
check::equal(Foo::get_count(), 2, "Should have 2 Foo objects");
check::equal(Foo::get_freearg_count(), 1, "freearg should have been used once");

// SWIG exception triggered and handled (return new object case).
try {
    trigger_internal_swig_exception("null", $b);
    check::fail("Expected exception not thrown");
} catch (Exception $e) {
}
check::equal(Foo::get_count(), 2, "Should have 2 Foo objects");
check::equal(Foo::get_freearg_count(), 2, "freearg should have been used twice");

// SWIG exception triggered and handled (return by value case).
try {
    trigger_internal_swig_exception("null");
    check::fail("Expected exception not thrown");
} catch (Exception $e) {
}
check::equal(Foo::get_count(), 2, "Should have 2 Foo objects");

File: Examples/test-suite/perl5/exception_memory_leak_runme.pl
#!/usr/bin/perl
use strict;
use warnings;
use Test::More tests => 11;
BEGIN { use_ok('exception_memory_leak') }
require_ok('exception_memory_leak');

my $a = exception_memory_leak::Foo->new();
is(exception_memory_leak::Foo::get_count(), 1, "Should have 1 Foo objects");
my $b = exception_memory_leak::Foo->new();
is(exception_memory_leak::Foo::get_count(), 2, "Should have 2 Foo objects");

# Normal behaviour
exception_memory_leak::trigger_internal_swig_exception("no problem", $a);
is(exception_memory_leak::Foo::get_count(), 2, "Should have 2 Foo objects");
is(exception_memory_leak::Foo::get_freearg_count(), 1, "freearg should have been used once");

# SWIG exception triggered and handled (return new object case).
eval {
  exception_memory_leak::trigger_internal_swig_exception("null", $b);
};
isnt($@, '', "Should throw exception");
is(exception_memory_leak::Foo::get_count(), 2, "Should have 2 Foo objects");
is(exception_memory_leak::Foo::get_freearg_count(), 2, "freearg should have been used twice");

# SWIG exception triggered and handled (return by value case).
eval {
  exception_memory_leak::trigger_internal_swig_exception("null");
};
isnt($@, '', "Should throw exception");
is(exception_memory_leak::Foo::get_count(), 2, "Should have 2 Foo objects");
